source/sympp/node/function/pow.cpp

```diff
@@ -34,7 +34,8 @@ namespace sympp {
     double pow::evaluate(const std::vector<uint8_t> &bool_values,
                          const std::vector<int> &int_values,
                          const std::vector<double> &double_values) const {
-        if (this->child_nodes_.front().compare(constant::e()) == 0) {
```

Comment:
Qual o motivo desta remoção?

Note que o compare agora é o equivalente ao starship operator. -1 quando lhs é menor, 0 quando são iguais, e +1 quando rhs é menor.

Reply:
A comparação em vermelho na linha 37, constava o erro bad_casting ( para Pow e para Log )

![Seleção_456](https://user-images.githubusercontent.com/57972585/96181918-1ec1b400-0f0b-11eb-93f6-c7d8e5fd8175.png)

Por isso havia alterada para  if(child_nodes_.back().begin()==(constant::e().begin())). 

Alterei a ordem da comparação : 

constant::e().compare(this->child_nodes_.front()) == 0 

Desta maneira funciona, uma vez que, compare espera um const sym e constant::e()  é um sym

Reply:
Entendi. Trocar a ordem parece paliativo. Evitar o erro encontrado mas não conserta. Não sei quem está lançando este bad cast, mas não era para estar ocorrendo pois um node_interface era para ter capacidade de ser comparado com qualquer sym. É como se a == b não funcionasse e b == a funcionasse. Não tem motivo. É necessário achar o problema.